### DISCUSSION TITLE
JSValue with std::wstring or hstring possible?

### DISCUSSION THREAD

**Discussion by creambyemute** [microsoft/react-native-windows#9399] [Q&A] [answered]
Hi there,

I've recently run into a Problem with react-native-fs windows implementation and paths/filenames which contain umlaute/accentuated character as these were passed std::string from JS --> Native.

I've created a PR (https://github.com/wwimmo/react-native-fs/pull/9) which changed most of the methods from receiving a `std::string` to a `std::wstring` but there were two methods that got passed a `JSValueObject`.

After a quick look I haven't found a way to get a value out of the JSValueObject as wstring/hstring, it only seems to support `.AsString()` which will return a std::string.

Is this not possible?

**Comment by asklar** [comment, marked as answer]
`JSValue::AsString` gives back the UTF-8 `std::string`, then `winrt::to_hstring` is supposed to convert that to a UCS-2 string; however in your PR you are creating the `std::filesystem::path` object from the UTF8 bytes instead of converting the UTF8 string to wstring then creating the `path` object with the `wstring`, can you try that?

**Reply by creambyemute** [reply]
Thanks, I'll try that!.

How about the other changes I did? Instead of passing ```std::string``` passing ```std::wstring``` from JS to Native? Should I use/pass ```winrt::hstring``` directly as well or is `std::wstring` fine?
